http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/CrtRequestExecutor.java

```diff
@@ -205,55 +91,28 @@ private void executeRequest(CrtRequestContext executionContext,
 
             // Submit the request on the connection
             crtConn.makeRequest(crtRequest, crtResponseHandler).activate();
-        } catch (HttpException e) {
-            crtConn.close();
-
-            Throwable toThrow = wrapWithIoExceptionIfRetryable(e);
-            requestFuture.completeExceptionally(toThrow);
-        } catch (IllegalStateException | CrtRuntimeException e) {
-            crtConn.close();
-            requestFuture.completeExceptionally(e);
```

Comment:
This is the previous code in sync code path. Note that the exception is not wrapped with IOException.